backend/app/role_assignments/global_/router.py

```diff
@@ -0,0 +1,118 @@
+from app.roles.service import ADMIN_UUID
+from app.users.schema import User
+
+router = APIRouter(prefix="/global")
+
+
+@router.get("")
+def list_assignments(
+    user_id: Optional[UUID] = None,
+    db: Session = Depends(get_db_session),
+    user: User = Depends(get_authenticated_user),
+) -> Sequence[RoleAssignmentResponse]:
+    return RoleAssignmentService(db=db, user=user).list_assignments(user_id=user_id)
+
+
+@router.post("")
+def create_assignment(
+    request: CreateRoleAssignment,
+    db: Session = Depends(get_db_session),
+    user: User = Depends(get_authenticated_user),
+) -> RoleAssignmentResponse:
+    if (role_id := request.role_id) == "admin":
+        role_id = ADMIN_UUID
+    return RoleAssignmentService(db=db, user=user).create_assignment(
+        RoleAssignmentRequest(user_id=request.user_id, role_id=role_id)
+    )
+
+
+@router.delete("/{id}")
+def delete_assignment(
+    id: UUID,
+    background_tasks: BackgroundTasks,
+    db: Session = Depends(get_db_session),
+    user: User = Depends(get_authenticated_user),
+) -> None:
+    assignment = RoleAssignmentService(db=db, user=user).delete_assignment(id)
+
+    if assignment.decision is DecisionStatus.APPROVED:
+        background_tasks.add_task(
+            tasks.remove_assignment, AuthAssignment.from_global(assignment)
+        )
+    return None
+
+
+@router.patch("/{id}/decide")
+def decide_assignment(
+    id: UUID,
+    request: DecideRoleAssignment,
+    background_tasks: BackgroundTasks,
+    db: Session = Depends(get_db_session),
+    user: User = Depends(get_authenticated_user),
+) -> RoleAssignmentResponse:
+    service = RoleAssignmentService(db=db, user=user)
+    original = service.get_assignment(id)
+
+    if original.decision not in (DecisionStatus.PENDING, request.decision):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="This assignment was already decided",
+        )
+
+    assignment = service.update_assignment(
+        UpdateRoleAssignment(id=id, decision=request.decision)
+    )
+
+    if assignment.decision is DecisionStatus.APPROVED:
+        background_tasks.add_task(
+            tasks.add_assignment, AuthAssignment.from_global(assignment)
+        )
+
+    return assignment
+
+
+@router.patch("/{id}/role")
+def modify_assigned_role(
+    id: UUID,
+    request: ModifyRoleAssignment,
+    background_tasks: BackgroundTasks,
+    db: Session = Depends(get_db_session),
+    user: User = Depends(get_authenticated_user),
+) -> RoleAssignmentResponse:
+    service = RoleAssignmentService(db=db, user=user)
+    original_role = service.get_assignment(id).role_id
+
+    if (role_id := request.role_id) == "admin":
+        role_id = ADMIN_UUID
+    assignment = service.update_assignment(UpdateRoleAssignment(id=id, role_id=role_id))
+
+    if assignment.decision is DecisionStatus.APPROVED:
+        assert (
+            original_role is not None
```

Comment:
I'm not following with this logic? Why does the original_role have to be not None? Doesn't this need to be the new role?

Reply:
also, modifying to an unexisting role sounds like removing? Maybe I'm completely misinterpreting this